selectors.go

```diff
@@ -32,153 +29,215 @@ const (
 	FamilyAptos    = "aptos"
 )
 
-var selectorsMap = parseYml(selectorsYml)
-var testSelectorsMap = parseYml(testSelectorsYml)
-
-var evmChainIdToChainSelector = loadAllSelectors()
-var selectorToChainFamily = loadSelectorToFamilyMap()
+var chainIDToSelectorMapForFamily = make(map[string]map[string]uint64)
+var testChainIDToSelectorMapForFamily = make(map[string]map[string]uint64)
+var selectorToChainDetails = loadChainDetailsBySelector()
+var testSelectorToChainDetailsMap = loadTestChainDetailsBySelector()
 
-func loadAllSelectors() map[uint64]chainDetails {
-	output := make(map[uint64]chainDetails, len(selectorsMap)+len(testSelectorsMap))
-	for k, v := range selectorsMap {
-		output[k] = v
+func loadTestChainDetailsBySelector() map[uint64]ChainDetails {
+	type yamlData struct {
+		Selectors map[uint64]ChainDetails `yaml:"selectors"`
 	}
-	for k, v := range testSelectorsMap {
-		output[k] = v
+
+	var data yamlData
+	err := yaml.Unmarshal(testSelectorsYml, &data)
+	if err != nil {
+		panic(err)
 	}
-	return output
-}
 
-func loadSelectorToFamilyMap() map[uint64]string {
-	type familyDetails struct {
-		Family string `yaml:"family"`
-		Name   string `yaml:"name"`
+	for k, v := range data.Selectors {
+		if v.Family == "" {
+			continue
+		}
+
+		// update testChainIDToSelectorMapForFamily
+		_, exist := testChainIDToSelectorMapForFamily[v.Family]
+		if exist {
+			testChainIDToSelectorMapForFamily[v.Family][v.ChainID] = k
+		} else {
+			testChainIDToSelectorMapForFamily[v.Family] = make(map[string]uint64)
+		}
 	}
 
+	return data.Selectors
+}
+
+func loadChainDetailsBySelector() map[uint64]ChainDetails {
 	type yamlData struct {
-		SelectorFamilies map[uint64]familyDetails `yaml:"selector_families"`
+		Selectors map[uint64]ChainDetails `yaml:"selectors"`
 	}
 
 	var data yamlData
-	err := yaml.Unmarshal(selectorFamiliesYml, &data)
+	err := yaml.Unmarshal(selectorYml, &data)
 	if err != nil {
 		panic(err)
 	}
 
-	var selectorFamilies = make(map[uint64]string, len(data.SelectorFamilies))
-	for k, v := range data.SelectorFamilies {
-		selectorFamilies[k] = v.Family
+	for k, v := range data.Selectors {
+		if v.Family == "" {
+			continue
+		}
+
+		// update chainIDToSelectorMapForFamily
+		_, exist := chainIDToSelectorMapForFamily[v.Family]
+		if exist {
+			chainIDToSelectorMapForFamily[v.Family][v.ChainID] = k
+		} else {
+			chainIDToSelectorMapForFamily[v.Family] = make(map[string]uint64)
```

Comment:
You need to also call the below for this case right:
chainIDToSelectorMapForFamily[v.Family][v.ChainID] = k

Reply:
Oh yes, thanks